src/formats/onenote.ts

```diff
@@ -0,0 +1,467 @@
+		// so we need to close them using regex
+		const regex = /<object([^>]*)\/>/g;
+		const pageElement = parseHTML(pageHTML.replace(regex, '<object$1></object>'));
+
+		const objects: HTMLElement[] = pageElement.findAll('object');
+		const images: HTMLImageElement[] = pageElement.findAll('img') as HTMLImageElement[];
+		// Online videos are implemented as iframes, normal videos are just <object>s
+		const videos: HTMLIFrameElement[] = pageElement.findAll('iframe') as HTMLIFrameElement[];
+
+		objects.forEach(async (object) => {
+			let split: string[] = object.getAttribute('data-attachment')!.split('.');
+			const extension: string = split[split.length - 1];
+
+			// If the page contains an incompatible file and user doesn't want to import them, skip
+			if (!ATTACHMENT_EXTS.contains(extension) && !this.importIncompatibleAttachments) {
+				return;
+			}
+			else {
+				this.attachmentQueue.push({
+					name: object.getAttribute('data-attachment')!,
+					contentLocation: object.getAttribute('data')!,
+				});
+		
+				// Create a new <p> element with the Markdown-style link
+				const markdownLink = document.createElement('p');
+				markdownLink.innerText = `![[${object.getAttribute('data-attachment')}]]`;
+		
+				// Replace the <object> tag with the new <p> element
+				object.parentNode?.replaceChild(markdownLink, object);
+			}
+		});
+
+		images.forEach(async (image) => {
+			let split: string[] = image.getAttribute('data-fullres-src-type')!.split('/');
+			const extension: string = split[1];
+			// TODO: there may be a similar function in the Obsidian API but I couldn't find it
+			const currentDate = (new Date).toISOString().replace(/[-:.TZ]/g, '').substring(0, 14);
+			const fileName: string = `Exported image ${currentDate}.${extension}`;
+
+			this.attachmentQueue.push({
+				name: fileName,
+				contentLocation: image.getAttribute('data-fullres-src')!,
+			});
+
+			image.src = encodeURIComponent(fileName);
+			if(!image.alt) image.alt = 'Exported image';
+		});
+
+		videos.forEach(async (video) => {
+			// Obsidian only supports embedding YouTube videos, unlike OneNote
+			if(video.src.contains('youtube.com') || video.src.contains('youtu.be')) {
+				const embedNode = document.createTextNode(`![Embedded YouTube video](${video.src})`);
+				video.parentNode?.replaceChild(embedNode, video);
+			}
+			else {
+				// If it's any other website, convert to a basic link
+				const linkNode = document.createElement('a');
+				linkNode.href = video.src;
+				video.parentNode?.replaceChild(linkNode, video);
+			}
+		});
+		return pageElement;
+	}
+
+	// Downloads attachments from the attachmentQueue once the file has been created.
+	async fetchAttachmentQueue(progress: ImportContext, currentFile: TFile) {
+		if (this.attachmentQueue.length >= 1) {
+			let attachmentPath: string = (await this.getOutputFolder())!.path + '/OneNote Attachments';
```

Comment:
this.getOutputFolder() should only be called once and validated before the importer starts it work - then the referenced TFolder should be passed.